### ISSUE TITLE
Bug(consistent-schema-var-name): Flags variable name `schema` as violation 

### ISSUE THREAD

**Issue by toto6038** [marcalexiei/eslint-plugin-zod#324]
### 📦 Affected package

eslint-plugin-zod

### 🧩 Description

The `consistent-schema-var-name` rule currently flags a variable named exactly `schema` and requires it to be renamed to `schemaSchema`, which is redundant and hurts readability.

### 🪜 How to reproduce

```ts
import * as z from 'zod'

// Flagged by the rule
const schema = z.object({ ... })
// Suggested auto-fix — but this looks odd
const schemaSchema = z.object({ ... })
```

### ✅ Expected behavior

I expected **no rule violation** since the bare variable name `schema` perfectly communicates its purpose as a Zod schema and technically _is_ the required suffix.

### 💻 System info

```shell
Binaries:
    Node: 24.11.1 - /usr/local/bin/node
    npm: 11.6.2 - /usr/local/bin/npm
    pnpm: 10.14.0 - /usr/local/bin/pnpm
  npmPackages:
    eslint: ^10.4.0 => 10.4.0 
    eslint-plugin-zod: ^4.5.2 => 4.5.2
```

### ✒️ Additional context

I am willing to submit PR if we agree it's an unhandled edge case :)

**Comment by marcalexiei** [comment]
Hello, thanks for opening the issue!

I agree, a name that matches the configured suffix alone shouldn't be flagged.

If you open a PR, please also handle the `prefix` side: when `before` is configured, a name equal to it should be accepted too.

### Proposed behaviour

Skip the report when `name.toLowerCase()` equals `before.toLowerCase()` or `after.toLowerCase()`.
The `before` / `after` options should also require at least 1 character.

The `<before><after>` concatenation case (`mySchema`) already passes, the existing `prefix` + `suffix` check, so no extra logic is needed.

**Comment by toto6038** [comment]
Hi, thanks for the quick feedback! I'd to discuss the `before` / `after` handling more carefully before opening PR.

If we skip reporting when `name.toLowerCase()` equals `before.toLowerCase()` or `after.toLowerCase()`, we'd create a case-insensitive acceptance path that doesn't exist anywhere else in the rule.

For example, with `{ before: 'Foo', after: '' }`:

- `foo` ✅ (case-insensitive bare-token match)
- `Foo` ✅ (case-insensitive bare-token match)
- `fooBar` ❌ (fails case-sensitive `startsWith('Foo')`)
- `FooBar` ✅ (passes case-sensitive `startsWith('Foo')`)

So bare tokens become case-insensitive, but longer names don't—a mismatch in rule semantics.

I suggest the fix can be narrowed: keep normal prefix/suffix checks case-sensitive, and only add a special bare-token exemption when **at least one** of `before` / `after` is an empty string:

- `{ before: '', after: 'Schema' }` → accepts `schema` (bare token) and `mySchema` (suffix match)
- `{ before: 'Foo', after: '' }` → accepts `Foo` and `foo` (both bare-token, case-insensitive) and `FooBar` (prefix match, case-sensitive)
- `{ before: 'Foo', after: 'Schema' }` → requires normal case-sensitive `FooxxxSchema` pattern (no bare-token exemption)

This way the rule is predictable: bare tokens are case-insensitive escapes **only** when one side is empty, but normal matching always respects casing.

**Comment by marcalexiei** [comment]
Good point on the semantic mismatch:
mixing case-insensitive bare-token matching with case-sensitive prefix/suffix checks when both sides are configured would be confusing.

Let's go with your narrowed proposal: the case-insensitive exemption applies only when one of `before` / `after` is omitted.
When both are configured, the rule keeps its current case-sensitive `before...after` behavior with no exemption.

Thanks for thinking this through!

**Comment by marcalexiei** [comment]
Available in
- [eslint-plugin-zod@4.5.3](https://github.com/marcalexiei/eslint-zod/releases/tag/eslint-plugin-zod%404.5.3)
- [eslint-plugin-zod-mini@1.2.1](https://github.com/marcalexiei/eslint-zod/releases/tag/eslint-plugin-zod-mini%401.2.1)
